### ISSUE TITLE
Regression in v7.25.0

### ISSUE THREAD

**Issue by delight** [pnpm/pnpm#5945]
### pnpm version:

7.25.0

### Code to reproduce the issue:

package.json with a (dev)-dependency on Supabase CLI: (version does not really matter):
		"supabase": "^1.33.0",

pnpm i 

### Expected behavior:
Download /  install dependencies

### Actual behavior:

 ENOENT  ENOENT: no such file or directory, chmod '${project_folder}/node_modules/supabase/bin/supabase'

### Additional information:
Works with pnpm@7.24.3 and earlier.

 - `node -v` prints:
 v18.13.0
but also breaks on v16.*

 - Windows, macOS, or Linux?:
 macOS and Linux

**Comment by zkochan** [comment]
Maybe it was caused by this PR: https://github.com/pnpm/pnpm/pull/5913

**Comment by bvanjoi** [comment]
More descriptions about recurrence:

- pnpm version: `7.25.0`
- macos x86

```npmrc
# <project>/.npmrc

node-linker="hoisted"
store-dir=".pnpm-store"
```

```json
// package.json

{
  "dependencies": {
    "supabase": "1.33.0"
  }
}
```

<img width="684" alt="image" src="https://user-images.githubusercontent.com/30187863/213212863-d2f26868-5db4-481d-867b-5db8ab7db6cc.png">

**Comment by spotvin42** [comment]
Hello, I have the same issue, needs to stay open.

**Comment by zkochan** [comment]
The fix to the issue was released in v7.25.1
